domains/certificates/RegistryConnector/Worker/EventHandlers/CertificateCreatedEventHandler.cs

```diff
@@ -0,0 +1,132 @@
+    private readonly ILogger<CertificateCreatedEventHandler> logger;
+    private readonly IKeyGenerator keyGenerator;
+    private readonly ProjectOriginRegistryOptions projectOriginRegistryOptions;
+
+    public CertificateCreatedEventHandler(RegistryService.RegistryServiceClient client,
+        ILogger<CertificateCreatedEventHandler> logger,
+        IKeyGenerator keyGenerator,
+        IOptions<ProjectOriginRegistryOptions> projectOriginRegistryOptions)
+    {
+        this.client = client;
+        this.logger = logger;
+        this.keyGenerator = keyGenerator;
+        this.projectOriginRegistryOptions = projectOriginRegistryOptions.Value;
+    }
+
+    public async Task Consume(ConsumeContext<CertificateCreatedEvent> context)
+    {
+        logger.LogInformation("Issuing to Registry for certificate id {certificateId}.", context.Message.CertificateId);
+
+        var message = context.Message;
+
+        var commitment = new SecretCommitmentInfo(message.Quantity);
+
+        var (ownerPublicKey, issuerKey) = keyGenerator.GenerateKeyInfo(message.Quantity,
+            message.WalletPublicKey, message.WalletDepositEndpointPosition, message.GridArea);
+
+        Transaction transaction;
+        if (message.MeteringPointType == MeteringPointType.Production)
+        {
+            var issuedEvent = Registry.CreateIssuedEventForProduction(
+                projectOriginRegistryOptions.RegistryName,
+                message.CertificateId,
+                message.Period.ToDateInterval(),
+                message.GridArea,
+                message.Gsrn.Value,
+                message.Technology!.TechCode,
+                message.Technology.FuelCode,
+                commitment,
+                ownerPublicKey);
+
+            transaction = issuedEvent.CreateTransaction(issuerKey);
+        }
+        else if (message.MeteringPointType == MeteringPointType.Consumption)
+        {
+            var issuedEvent = Registry.CreateIssuedEventForConsumption(
+                projectOriginRegistryOptions.RegistryName,
+                message.CertificateId,
+                message.Period.ToDateInterval(),
+                message.GridArea,
+                message.Gsrn.Value,
+                commitment,
+                ownerPublicKey);
+
+            transaction = issuedEvent.CreateTransaction(issuerKey);
+        }
+        else
+            throw new CertificateDomainException(string.Format("Unsupported metering point type {0} for certificateId {1}",
+                message.MeteringPointType, message.CertificateId));
+
+        var request = new SendTransactionsRequest();
+        request.Transactions.Add(transaction);
+
+        // The Registry is idempotent wrt. sending the same Transaction. Re-sending the transaction
+        // after it has been committed, will not change the status of the initial transaction
+        await client.SendTransactionsAsync(request);
+
+        await context.Publish(new CertificateSentToRegistryEvent
```

Comment:
Didnt we talk about using Send instead so where it was sent to?

Reply:
Send is for commands and publish is for events. I chose events because the steps are async and because they are recorded facts